### DISCUSSION TITLE
I deleted and recovered "not-found.tsx", then suddenly Vercel shows their own 404 page instead of my "not-found.tsx".

### DISCUSSION THREAD

**Discussion by DongeunPaeng** [vercel/next.js#68398] [Help]
### Summary

In my Next.js project, I mistakenly deleted the not-found.tsx file and deployed the project to Vercel. Later, I added the not-found.tsx file back to the project and redeployed it. Between the two deploy, nothing has changed. I didn't even type a new letter to any file.

Despite restoring the project to its original state, Vercel is not displaying the custom 404 page rendered by not-found.tsx. Instead, I see Vercel's 404 page.

![image](https://github.com/user-attachments/assets/c0407659-3e96-4bcd-9f19-b3dbfc48ecca)

I tried:

1) Emptying cache in Vercel's project settings page.
2) Trying "next build" -> "next start" in my local environment and it shows the custom 404 page properly.
3) Waiting 7 days in order to see if Vercel keeps any Edge Network resources.
4) Making a slight change in "not-found.tsx" file and deployed again.
5) Checking Vercel's build log. (see below what it printed)
```
Route (pages)                              Size     First Load JS
─ ○ /404                                   181 B          80.5 kB
```

I'm curious what I missed.

Thank you, community!

(My project structure and the not-found.tsx file)
![image](https://github.com/user-attachments/assets/339f5eca-a7cb-434d-9e84-82098c49f4c5)

**Comment by Pearces1997** [comment]
Same issue, deleted vercel project and deployed on different provider. 404 on previously working page served from vercel... The vercel project that I deleted still messing with DNS records... Have not been able to fix even redeploy won't overwrite the zombie project.
